src/components/input/math-input.js

```diff
@@ -218,6 +221,20 @@ class MathInput extends React.Component {
         this._keypadBounds = keypadNode.getBoundingClientRect();
     };
 
+    _keepInputFocus = event => {
+        if (!this.state.focused) {
+          return;
+        }
+    
+        if (event.relatedTarget === null) {
+          event.preventDefault();
+          this.inputRef.focus();
```

Comment:
This only happens if THIS input is focused. There's a check for that at the start of this block

Reply:
What I'm getting from this code is that - if a user is inputting via touching the keypad, either with their finger/finger alternative or with a mouse, we keep the focus on the input field itself despite the fact that the user is interacting with the keypad because we're treating the keypad as if it were an external keyboard. Is that right?

Some documentation here explaining the focus flow would really help as it's not a common pattern and a little difficult to reason about.

The only thing I'm concerned about it what this flow is like if you have a screen reader on. Can a user still navigate through the keypad keys without the focus being pulled back to the input field?

Reply:
I believe the flow still works out ok (at least in my testing!). I've added a bunch of comments to make more clear what's going on here. Thank you for the suggestion.

Reply:
I was also wondering what happens if there are multiple widgets using math-input in an exercise question? Is it still working as expected with the input the user is currently interacting.